src/main/java/com/synopsys/integration/blackduck/service/bucket/BlackDuckBucketService.java

```diff
@@ -59,28 +61,30 @@ public BlackDuckBucket startTheBucket(final List<UriSingleResponse<? extends Bla
         return blackDuckBucket;
     }
 
-    public <T extends BlackDuckResponse> void addToTheBucket(final BlackDuckBucket blackDuckBucket, final String uri, final Class<T> responseClass) throws IntegrationException {
-        final List<UriSingleResponse<? extends BlackDuckResponse>> uriSingleResponses = new ArrayList<>();
-        uriSingleResponses.add(new UriSingleResponse<>(uri, responseClass));
-        addToTheBucket(blackDuckBucket, uriSingleResponses);
+    public Future<Optional<? extends BlackDuckResponse>> addToTheBucket(final BlackDuckBucket blackDuckBucket, final String uri, final Class<? extends BlackDuckResponse> responseClass) throws IntegrationException {
```

Comment:
Moving to ? from T - is that lossy at all? Is the binding of the returned class the same as the method parameter? I thought each ? got its own binding.